### PR TITLE
feat(github): Optional ETag handling for PR cache

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Additional flag to turn on/off `ETag` checks
- Turn off `ETag` handling for remote repo cache

## Context

- Ref: #15118

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
When repo cache is remote, this will make us to request at least one page of PRs per run, but otherwise we'll be pushing to remote cache each run and trigger `ETag` update anyway.

**Comment by rarkins** [comment]
Is this a user visible change?

**Comment by zharinov** [comment]
No, it's not. It starts to work with `repositoryCache=remote`, but this isn't possible to set for now.

File: lib/modules/platform/github/pr.ts [hunk 1]
```diff
@@ -102,11 +123,18 @@ export async function getPrCache(
         );
       }
 
+      if (!handleEtag) {
+        page.forEach((ghPr) => {
```
Comment (viceice) [inline lib/modules/platform/github/pr.ts:127]: can we use for of loop instead?

**Review by viceice — DISMISSED** [review]

**Comment by rarkins** [comment]
Needs deconflicting

**Comment by zharinov** [comment]
Superseded by #15292

**Pull request closed without merging**